### PR TITLE
Lwrp support

### PR DESCRIPTION
**Pull request by jimmy-jam**
Creating a branch to show LWRP support.

### CODE REVIEW

File: Assets/LWRPSupport/ARCoreBackgroundLWRP.shader [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿Shader "Unlit/ARCoreBackgroundLWRP"
```
Comment (tdmowrer) [inline Assets/LWRPSupport/ARCoreBackgroundLWRP.shader]: Please update this shader to match the current one for ARCore.

File: Assets/LWRPSupport/ARKitLWRPShader.shader [hunk 2]
```diff
@@ -0,0 +1,77 @@
+Shader "Unlit/ARKitLWRP"
+{
+	Properties
+	{
+    	_textureY ("TextureY", 2D) = "white" {}
```
Comment (tdmowrer) [inline Assets/LWRPSupport/ARKitLWRPShader.shader]: spacing

File: Assets/LWRPSupport/LWRPBeforeCameraRender.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+		public void SetupBlitMaterial(Material material)
+		{
+			this.blitMaterial = material;
+		}
+
+		public void ExecuteBeforeCameraRender(ScriptableRenderContext context, Camera currentCamera, LightweightPipeline.PipelineSettings pipelineSettings,
+			ScriptableRenderer currentRenderer)
+		{
+			if (blitMaterial == null) return;
```
Comment (tdmowrer) [inline Assets/LWRPSupport/LWRPBeforeCameraRender.cs]: `return` on newline please

File: Assets/LWRPSupport/LWRPBeforeCameraRender.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+using UnityEngine.Rendering;
+using UnityEngine.Experimental.Rendering.LightweightPipeline;
+
+namespace UnityEngine.XR.ARFoundation
+{
+	public class LWRPBeforeCameraRender : MonoBehaviour, LightweightPipeline.IBeforeCameraRender
+	{
+		const string k_ARBlitTag = "ARBackground Blit Pass";
+
+		Material blitMaterial { get; set; }
```
Comment (tdmowrer) [inline Assets/LWRPSupport/LWRPBeforeCameraRender.cs]: Any reason this shouldn't be public? Then you could get rid of `SetupBlitMaterial` below.

File: Assets/LWRPSupport/LWRPBackgroundRenderer.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
+﻿
+namespace UnityEngine.XR.ARFoundation.LWRPSupport
+{
+
+    public class LWRPBackgroundRenderer : ARFoundationBackgroundRenderer
+    {
+        CameraClearFlags m_SavedClearFlags;
+        LWRPBeforeCameraRender m_LWRPBeforeCameraRender;
+
+        protected override bool EnableARBackgroundRendering()
```
Comment (tdmowrer) [inline Assets/LWRPSupport/LWRPBackgroundRenderer.cs]: This still feels like it should be broken into two methods.

File: Assets/LWRPSupport/LWRPBackroundRendererAsset.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    [CreateAssetMenu(fileName = "LWRPBackroundRendererAsset", menuName = "XR/LWRPBackroundRendererAsset")]
+    public class LWRPBackroundRendererAsset : ARBackgroundRendererAsset
+    {
+        /// <summary>
+        /// we're going to reference all materials that we want to use so that they get built into the project
+        /// </summary>
+        [SerializeField]
+        Material[] m_MaterialsUsed;
+        
+        public override ARFoundationBackgroundRenderer CreateARBackgroundRenderer()
```
Comment (tdmowrer) [inline Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]: Should this be `static`?

File: Assets/LWRPSupport/LWRPBackroundRendererAsset.cs [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            var useRenderPipeline = GraphicsSettings.renderPipelineAsset != null;
```
Comment (tdmowrer) [inline Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]: This is used a couple of places. Perhaps it should be a (private) property?

File: Assets/LWRPSupport/LWRPBackroundRendererAsset.cs (hunk 6 above)
Comment (tdmowrer) [reply Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]: Other impl could require member access; this one just doesn't happen to

File: Assets/LWRPSupport/LWRPBackroundRendererAsset.cs (hunk 7 above)
Comment (jimmy-jam) [reply Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]: does static private property work?

Comment (jimmy-jam) [reply Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]:
```
       static bool useRenderPipeline {  get { return GraphicsSettings.renderPipelineAsset != null; } }
 ```

File: Assets/LWRPSupport/ARCoreBackgroundLWRP.shader (hunk 1 above)
Comment (jimmy-jam) [reply Assets/LWRPSupport/ARCoreBackgroundLWRP.shader]: check against preview.19 and all i saw was a Cull Off which I added (and converted tabs to spaces) - let me know if there was something else.

File: Assets/LWRPSupport/LWRPBackroundRendererAsset.cs (hunk 7 above)
Comment (tdmowrer) [reply Assets/LWRPSupport/LWRPBackroundRendererAsset.cs]: lgtm

File: Assets/LWRPSupport/ARCoreBackgroundLWRP.shader (hunk 1 above)
Comment (tdmowrer) [reply Assets/LWRPSupport/ARCoreBackgroundLWRP.shader]: That's what stood out to me as different; wanted to make sure there was nothing else; sounds like that was it.

**Pull request merged**